src/test/scala/edu/berkeley/cs/rise/opaque/OpaqueOperatorTests.scala

```diff
@@ -375,17 +399,28 @@ trait OpaqueOperatorTests extends FunSuite with BeforeAndAfterAll { self =>
   }
 
   testAgainstSpark("aggregate average") { securityLevel =>
-    val data = for (i <- 0 until 256) yield (i, abc(i), i.toDouble)
+    val data = (0 until 256).map{ i =>
+      if (i % 3 == 0 || (i + 1) % 6 == 0)
+        (i, abc(i), None)
+      else
+        (i, abc(i), Some(i.toDouble))
+    }.toSeq
     val words = makeDF(data, securityLevel, "id", "category", "price")
+    words.setNullableStateOfColumn("price", true)
```

Comment:
Have the aggregate {average, count, sum} tests all been modified such that one of the columns is null? Maybe for each of {average, count, sum}, we can have one test where none of the columns are null, and one test where one column is null (like you have now).

Reply:
The tests were modified so that for the three of the groups, one group has no nulls, one has all nulls, and one has a mix of nulls and non-null values. Is this enough?

Reply:
Oh perfect! This is great.